tensorflow/python/framework/graph_util_impl.py

```diff
@@ -123,6 +123,11 @@ def extract_sub_graph(graph_def, dest_nodes):
   if not isinstance(graph_def, graph_pb2.GraphDef):
     raise TypeError("graph_def must be a graph_pb2.GraphDef proto.")
 
+  # Github 13047, cast single string to list
+  if isinstance(dest_nodes, str):
```

Comment:
Use six.string_types instead of str. See [saver.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/saver.py#L583) for an example.

Reply:
Hi, @reedwm . If string is invalid, how about raising TypeError if only dest_nodes is not list? Please take a look at the latest commit, thanks.